src/init.c

```diff
@@ -112,10 +112,11 @@ void (*_dispatch_end_NSAutoReleasePool)(void *);
 #endif
 
 #if DISPATCH_USE_THREAD_LOCAL_STORAGE
-__thread struct dispatch_tsd __dispatch_tsd;
 #if defined(_WIN32)
+__declspec(thread) struct dispatch_tsd __dispatch_tsd;
```

Comment:
would using `_Thread_local` work instead? I would assume it maps to the right thing on both platforms, and libdispatch otherwise already requires C11

Reply:
MSVC actually doesn't support C11 threads, so using `__declspec(thread)` is the canonical method of declaring thread-local variables in Windows code. I agree that using `_Thread_local` everywhere would be best if we don't care about building with MSVC (I don't).

Reply:
by MSVC you mean the MSFT C compiler as opposed to clang, correct? Not MSFT Visual Studio?

libdispatch needs clang (which is I'm told supported on Windows nowadays?) for blocks support anyway (and more). So yeah I don't think we care about MSVC in this instance, if `_Thread_local` does the job, we should use it, `__thread` really is a gcc-ism and C11 has the right thing.

Reply:
>by MSVC you mean the MSFT C compiler as opposed to clang, correct?

Right.

I updated this to use `_Thread_local` now. Thanks for the guidance.